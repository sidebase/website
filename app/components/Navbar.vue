<script setup lang="ts">
import Logo from '~/assets/branding/Sidebase_logo_dark_icon_font_for_dark_bg.svg'
</script>

<template>
  <div class="w-full flex items-center justify-between lg:container mx-auto">
    <div class="w-[250px] h-[100px] overflow-hidden flex items-center justify-center">
      <Logo width="250px" height="200px" />
    </div>
    <div class="flex items-center space-x-5">
      <a class="cursor-pointer text-SideBase-green hover:text-white transition-colors text-lg font-light">
        <i class="pi pi-github pt-0.5" style="font-size: 1.25rem" />
      </a>
      <a class="cursor-pointer text-SideBase-green hover:text-white transition-colors text-lg font-light">
        <i class="pi pi-twitter pt-1" style="font-size: 1.25rem" />
      </a>
    </div>
  </div>
</template>
