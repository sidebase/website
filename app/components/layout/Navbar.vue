<script setup lang="ts">
import Button from 'primevue/button'
import Logo from '~/assets/branding/Sidebase_logo_dark_icon_font_for_dark_bg.svg'
</script>

<template>
  <div class="w-full flex items-center justify-between container mx-auto px-6 pt-2">
    <div class="-ml-2 w-[150px] h-[50px] lg:w-[250px] lg:h-[100px] overflow-hidden flex items-center justify-center">
      <Logo />
    </div>
    <div class="flex items-center space-x-5">
      <a href="https://github.com/sidestream-tech/sidebase" target="_blank" class="cursor-pointer text-SideBase-green hover:text-white transition-colors text-lg font-light">
        <i class="pi pi-github pt-0.5" style="font-size: 1.25rem" />
      </a>
      <a href="https://twitter.com/sidebase_io" target="_blank" class="cursor-pointer text-SideBase-green hover:text-white transition-colors text-lg font-light">
        <i class="pi pi-twitter pt-1" style="font-size: 1.25rem" />
      </a>
      <!--
      <Button class="stars p-0">
        <i class="pi pi-star-fill px-2" />
        <span class="px-3">231</span>
      </Button>
      -->
    </div>
  </div>
</template>

<style lang="scss" scoped>
.p-button.stars {
  background: linear-gradient(to left, var(--green-400) 50%, var(--green-500) 50%);
  background-size: 200% 100%;
  background-position: right bottom;
  transition: background-position 0.5s ease-out;
  color: #fff;
  padding: 0;
  border: 0 solid;
}
.p-button.stars:hover {
  background-position: left bottom;
}
.p-button.stars i {
  background-color: var(--green-500);
  line-height: 2.25rem;
}
.p-button.stars:focus {
  box-shadow: 0 0 0 1px var(--green-600);
}
</style>
