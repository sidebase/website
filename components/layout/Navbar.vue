<script lang="ts" setup>
import SidebaseLogoDark from '~/assets/sidebase_logo_dark_just_icon_for_dark_bg.svg'
</script>

<template>
  <nav class="bg-pickled-bluewood-800 px-2 py-2.5 sm:px-4">
    <div class="container mx-auto flex flex-wrap items-center justify-between">
      <nuxt-link to="/" class="flex items-center">
        <SidebaseLogoDark class="mr-3 h-6 sm:h-8" alt="sidebase logo" />
        <span class="font-CooperHewitt dark:text-sidebase-green-400 self-center whitespace-nowrap pt-1 text-xl font-light">sidebase</span>
      </nuxt-link>
      <a href="https://github.com/sidestream-tech/sidebase" target="_blank" class="ml-3 inline-flex items-center rounded-lg p-2 text-sm text-gray-400 hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-gray-600 md:hidden">
        Github
      </a>
      <div class="hidden w-full md:block md:w-auto">
        <ul class="mt-4 mb-0 flex flex-col rounded-lg p-4 md:mt-0 md:flex-row md:space-x-8 md:border-0 md:text-sm md:font-medium">
          <li>
            <nuxt-link to="/" class="block rounded py-2 pr-4 pl-3 text-gray-400 hover:text-white md:border-0 md:p-0">
              Home
            </nuxt-link>
          </li>
          <li>
            <nuxt-link to="/getting-started" class="block rounded py-2 pr-4 pl-3 text-gray-400 hover:text-white md:border-0 md:p-0">
              Getting started
            </nuxt-link>
          </li>
          <li>
            <nuxt-link to="/why" class="block rounded py-2 pr-4 pl-3 text-gray-400 hover:text-white md:border-0 md:p-0">
              Why sidebase?
            </nuxt-link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</template>
