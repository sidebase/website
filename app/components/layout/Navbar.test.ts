import { describe, expect, test } from 'vitest'
import Navbar from './Navbar.vue'
import { render } from '~/tests/testingLibraryVue'

describe('Navbar', () => {
  test('mounts and is still the same', () => {
    const { html } = render(Navbar)
    expect(html()).toMatchSnapshot()
  })
})
