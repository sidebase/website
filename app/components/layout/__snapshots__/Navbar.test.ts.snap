// Vitest Snapshot v1

exports[`Navbar > mounts and is still the same 1`] = `
"<div class=\\"w-full flex items-center justify-between container mx-auto px-6 pt-2\\" data-v-c7ed1fd0=\\"\\">
  <div class=\\"-ml-2 w-[150px] h-[50px] lg:w-[250px] lg:h-[100px] overflow-hidden flex items-center justify-center\\" data-v-c7ed1fd0=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"500\\" viewBox=\\"0 0 375 375\\" height=\\"500\\" version=\\"1.0\\" data-v-c7ed1fd0=\\"\\">
      <defs>
        <clipPath id=\\"a\\">
          <path d=\\"M56.27 146.863h69v59.25h-69Zm0 0\\"></path>
        </clipPath>
        <clipPath id=\\"b\\">
          <path d=\\"M37.5 157.816h69v59.25h-69Zm0 0\\"></path>
        </clipPath>
        <clipPath id=\\"c\\">
          <path d=\\"M56.27 168.773h69v59.25h-69Zm0 0\\"></path>
        </clipPath>
      </defs>
      <path d=\\"M149.409 208.914c-6.41 0-8.074-3.442-8.074-6.88h-2.086v.208c0 3.804 2.398 8.648 10.16 8.648 6.46 0 10.16-3.281 10.16-7.71v-.208c0-4.793-2.45-6.98-9.379-7.504-6.93-.57-8.441-2.394-8.441-5.68 0-3.593 2.867-5.574 7.66-5.574 6.043 0 7.816 2.868 7.816 7.399h2.082v-.207c0-5.211-2.55-9.172-9.898-9.172-6.461 0-9.848 3.23-9.848 7.453v.258c0 3.754 1.774 6.566 9.067 7.242 6.093.523 8.753 1.46 8.753 5.89 0 3.594-2.867 5.837-7.972 5.837ZM167.123 210.37h2.086v-27.613h-2.086Zm2.605-38.296h-3.125v3.129h3.125ZM186.922 182.234c-7.555 0-10.68 6.254-10.68 13.808v1.043c0 7.555 3.125 13.805 10.68 13.805 4.375 0 6.98-2.082 8.336-4.426l.52 3.907h1.566l-.262-7.817v-33.867h-2.086v17.508c-1.457-2.137-3.957-3.961-8.074-3.961Zm8.074 21.883c-1.3 2.086-3.594 4.797-7.812 4.797-5.891 0-8.754-4.641-8.754-11.829v-1.043c0-7.19 2.863-11.828 8.754-11.828 4.218 0 6.511 2.711 7.812 4.793ZM225.998 197.554v-1.512c0-7.554-3.336-13.808-11.203-13.808-7.867 0-11.2 6.254-11.2 13.808v1.043c0 7.555 3.387 13.805 11.461 13.805 7.711 0 10.942-5.207 10.942-9.168v-.21h-2.082c0 3.386-2.45 7.402-8.86 7.402-6.41 0-9.171-4.586-9.273-11.36Zm-11.203-13.34c6.047 0 8.86 4.586 9.015 11.36h-18.027c.102-6.774 2.918-11.36 9.012-11.36ZM234.594 168.687h-2.082v33.867l-.262 7.817h1.562l.47-3.907c1.405 2.344 3.96 4.426 8.39 4.426 7.554 0 10.68-6.25 10.68-13.805v-1.043c0-7.554-3.126-13.808-10.68-13.808-4.117 0-6.672 1.824-8.078 3.96Zm0 35.43v-15.11c1.304-2.082 3.597-4.793 7.816-4.793 5.887 0 8.754 4.637 8.754 11.828v1.043c0 7.188-2.867 11.829-8.754 11.829-4.219 0-6.512-2.711-7.816-4.797ZM277.318 210.37h1.563l-.262-7.816v-10.16c0-5.887-2.188-10.16-9.375-10.16-7.817 0-9.38 5.21-9.38 9.172v.468h2.083c0-4.167 1.305-7.66 7.297-7.66 5.574 0 7.293 2.813 7.293 8.18v1.043l-8.961 1.406c-4.067.625-8.754 2.242-8.754 8.235v.52c0 4.429 3.125 7.292 7.816 7.292 5.207 0 8.387-2.293 10.16-4.687Zm-10.422-1.456c-4.168 0-5.992-2.555-5.992-5.317v-.52c0-3.597 2.605-5.523 6.516-6.203l9.117-1.507v8.336c-.992 1.77-3.492 5.21-9.64 5.21ZM295.814 208.914c-6.41 0-8.074-3.442-8.074-6.88h-2.086v.208c0 3.804 2.398 8.648 10.16 8.648 6.461 0 10.16-3.281 10.16-7.71v-.208c0-4.793-2.449-6.98-9.379-7.504-6.93-.57-8.441-2.394-8.441-5.68 0-3.593 2.867-5.574 7.66-5.574 6.043 0 7.817 2.868 7.817 7.399h2.082v-.207c0-5.211-2.551-9.172-9.899-9.172-6.46 0-9.848 3.23-9.848 7.453v.258c0 3.754 1.774 6.566 9.067 7.242 6.094.523 8.754 1.46 8.754 5.89 0 3.594-2.867 5.837-7.973 5.837ZM334.63 197.554v-1.512c0-7.554-3.336-13.808-11.204-13.808-7.867 0-11.199 6.254-11.199 13.808v1.043c0 7.555 3.387 13.805 11.461 13.805 7.711 0 10.942-5.207 10.942-9.168v-.21h-2.082c0 3.386-2.45 7.402-8.86 7.402-6.41 0-9.172-4.586-9.273-11.36Zm-11.204-13.34c6.047 0 8.86 4.586 9.016 11.36h-18.027c.101-6.774 2.918-11.36 9.011-11.36Zm0 0\\" fill=\\"#fff\\"></path>
      <g clip-path=\\"url(#a)\\">
        <path fill=\\"#fff\\" d=\\"M122.29 206.113H59.25a2.787 2.787 0 0 1-2.445-1.406 2.78 2.78 0 0 1 0-2.809l30.875-53.253a3.545 3.545 0 0 1 3.09-1.782c1.293 0 2.445.66 3.09 1.782l30.874 53.27c.508.874.508 1.929 0 2.804a2.82 2.82 0 0 1-2.445 1.394Zm-63.794-3.242a.831.831 0 0 0 0 .875c.082.137.3.438.754.438h63.023c.454 0 .688-.301.758-.438a.831.831 0 0 0 0-.875l-30.86-53.27a1.59 1.59 0 0 0-1.401-.808c-.59 0-1.114.3-1.403.809Zm0 0\\"></path>
      </g>
      <g clip-path=\\"url(#b)\\">
        <path fill=\\"#585e63\\" d=\\"M103.52 217.066H40.48a2.787 2.787 0 0 1-2.445-1.406 2.772 2.772 0 0 1 0-2.805l30.875-53.257a3.545 3.545 0 0 1 3.09-1.782c1.29 0 2.445.66 3.09 1.782l30.875 53.27c.508.874.508 1.929 0 2.804a2.82 2.82 0 0 1-2.445 1.394Zm-63.793-3.242a.831.831 0 0 0 0 .875c.082.137.3.442.753.442h63.024a.847.847 0 0 0 .758-.442.831.831 0 0 0 0-.875l-30.86-53.27a1.59 1.59 0 0 0-1.402-.808c-.59 0-1.113.3-1.402.809Zm0 0\\"></path>
      </g>
      <g clip-path=\\"url(#c)\\">
        <path fill=\\"#5bf3c7\\" d=\\"M122.29 228.023H59.25a2.79 2.79 0 0 1-2.445-1.41 2.772 2.772 0 0 1 0-2.804L87.68 170.55a3.542 3.542 0 0 1 3.09-1.778c1.293 0 2.445.657 3.09 1.778l30.874 53.27c.508.874.508 1.929 0 2.804a2.823 2.823 0 0 1-2.445 1.398Zm-63.794-3.246a.831.831 0 0 0 0 .875c.082.141.3.442.754.442h63.023a.84.84 0 0 0 .758-.442.831.831 0 0 0 0-.875l-30.86-53.27a1.59 1.59 0 0 0-1.401-.808c-.59 0-1.114.301-1.403.809Zm0 0\\"></path>
      </g>
    </svg></div>
  <div class=\\"flex items-center space-x-5\\" data-v-c7ed1fd0=\\"\\"><a href=\\"https://github.com/sidestream-tech/sidebase\\" target=\\"_blank\\" class=\\"cursor-pointer text-SideBase-green hover:text-white transition-colors text-lg font-light\\" data-v-c7ed1fd0=\\"\\"><i class=\\"pi pi-github pt-0.5\\" style=\\"font-size:1.25rem;\\" data-v-c7ed1fd0=\\"\\"></i></a><a href=\\"https://twitter.com/sidebase_io\\" target=\\"_blank\\" class=\\"cursor-pointer text-SideBase-green hover:text-white transition-colors text-lg font-light\\" data-v-c7ed1fd0=\\"\\"><i class=\\"pi pi-twitter pt-1\\" style=\\"font-size:1.25rem;\\" data-v-c7ed1fd0=\\"\\"></i></a>
    <!--
      &lt;Button class=&quot;stars p-0&quot;&gt;
        &lt;i class=&quot;pi pi-star-fill px-2&quot; /&gt;
        &lt;span class=&quot;px-3&quot;&gt;231&lt;/span&gt;
      &lt;/Button&gt;
      -->
  </div>
</div>"
`;
